File: lang-java/src/test/resources/fixtures/cardinality-explosion/negative/rederived-per-outer-iteration.java
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class LocaleNameValidator {
    public void validateNames(List<String> locales) {
        for (String locale : locales) {
            Set<String> seenNames = new HashSet<>();
            List<String> namesInLocale = getNames(locale);
            for (String name : namesInLocale) {
                if (!seenNames.add(name)) {
                    throw new IllegalStateException("Duplicate name: " + name);
                }
            }
        }
    }

    private List<String> getNames(String locale) {
        return List.of();
    }
}

File: lang-kotlin/src/test/resources/fixtures/cardinality-explosion/negative/rederived-per-outer-iteration.kt
class LocaleNameValidator {
    fun validateNames(locales: List<String>) {
        for (locale in locales) {
            val seenNames = HashSet<String>()
            val namesInLocale = getNames(locale)
            for (name in namesInLocale) {
                if (!seenNames.add(name)) {
                    throw IllegalStateException("Duplicate name: $name")
                }
            }
        }
    }

    private fun getNames(locale: String): List<String> = emptyList()
}
